cloud/filestore/libs/storage/service/service_ut_sharding.cpp

```diff
@@ -4019,6 +4019,133 @@ Y_UNIT_TEST_SUITE(TStorageServiceShardingTest)
+        service.CreateFileStore(fsId, 20_MB / 4_KB);
+
+        // waiting for IndexTablet start after the restart triggered by
+        // configureshards
+        WaitForTabletStart(service);
+
+        TVector<TString> expected = {
+            fsId,
+            fsId + "_s1",
+            fsId + "_s2",
+            fsId + "_s3",
+            fsId + "_s4",
+            fsId + "_s5",
+        };
+        auto listing = service.ListFileStores();
+        auto fsIds = listing->Record.GetFileStores();
+        TVector<TString> ids(fsIds.begin(), fsIds.end());
+        Sort(ids);
+        UNIT_ASSERT_VALUES_EQUAL(expected, ids);
+
+        auto headers = service.InitSession(fsId, "client");
+
+        TVector<ui64> handles;
+        TVector<ui64> nodes;
+        TSet<ui32> shards;
+        for (ui32 i = 0; i < 5; ++i) {
+            auto createHandleResponse = service.CreateHandle(
+                headers,
+                fsId,
+                RootNodeId,
+                Sprintf("file%u", i),
+                TCreateHandleArgs::CREATE)->Record;
+
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
+
+            const auto handleId = createHandleResponse.GetHandle();
+            handles.push_back(handleId);
+        }
+
+        UNIT_ASSERT_VALUES_EQUAL(5, shards.size());
+
+        service.WriteData(
+            headers,
+            fsId,
+            nodes[0],
+            handles[0],
+            0,
+            TString(3_MB + 4_KB, 'a'));
+
+        service.WriteData(
+            headers,
+            fsId,
+            nodes[2],
+            handles[2],
+            0,
+            TString(3_MB + 4_KB, 'a'));
+
+        service.WriteData(
+            headers,
+            fsId,
+            nodes[4],
+            handles[4],
+            0,
+            TString(3_MB + 4_KB, 'a'));
+
+        // waiting for async stats aggregation from shards
+        // doing it before triggering another event to avoid DispatchEvents call
+        // which does a long busy-wait loop
+        env.GetRuntime().AdvanceCurrentTime(TDuration::Seconds(15));
+        // just triggering another event chain - doesn't matter which one
+        service.StatFileStore(headers, fsId);
+
+        TSet<ui32> emptyShards;
+        emptyShards.insert(ExtractShardNo(nodes[1]));
+        emptyShards.insert(ExtractShardNo(nodes[3]));
+
+        handles.clear();
+        nodes.clear();
+        shards.clear();
+
+        for (ui32 i = 0; i < 5; ++i) {
+            auto createHandleResponse = service.CreateHandle(
+                headers,
+                fsId,
+                RootNodeId,
+                Sprintf("file%u", 5 + i),
+                TCreateHandleArgs::CREATE)->Record;
+
+            const auto nodeId = createHandleResponse.GetNodeAttr().GetId();
+            shards.insert(ExtractShardNo(nodeId));
+            nodes.push_back(nodeId);
```

Comment:
do you want to test that you are getting different shard each time to validate that allocation rotates across available shards

Reply:
will be resolved in next pr